### ISSUE TITLE
Org Member Page - Domain Assignment Page

### ISSUE THREAD

**Issue by therealslimhsiehdy** [cisagov/manage.get.gov#2767]
### Issue description

As part of Member management, org admins will be able to change which domains a user has management permissions to them (User domain roles) or which they are invited to (domain invitations). On this page, users should be able to search and sort domains that a user is assigned to, prior to editing these assignments. Eventually, we will add in functionality so you can click a button and change those assigned domains (#2768 ), but for now we are just starting with being able to see the domains and having the button with no functionality.

### Acceptance criteria

* [Figma design here](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=7364-28508&t=T46HUmpFC0ZXDHHg-4) 
- [x] Clicking the "View" OR "Manage" icon on "Domain Management" section on the Manage member page takes you to the new "Domain assignments" page
- [x] If the user only has permissions to see User Management but **does not** have edit permissions, then do not show an "Edit domain assignments button"
- [x] If the user has both view and edit permissions for User Management, then show the "Edit domain assignments button"
- [x] Includes annotations 1 - 3
- [x] "Edit Domain Assignments button" should do nothing for now (it will redirect to a edit page later)
- [ ] If no domains are assigned, [see Figma for what to display text wise](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=8901-83623&t=T46HUmpFC0ZXDHHg-4). 
- [ ] If domains are assigned - have the domains in a table that is **sortable** by domain name and has **search** functionality. There is **no pagination** on this page.
- [x] Make sure to note that there is different search caption text ("search domains assigned to x" vs "search all domains")
- [x] The domains shown include domains the user is invited to and which they already have user domain roles for, but **only** domains within the current portfolio

### Not in scope
- [ ] Edit domain assignments button does NOT function

### Links to other issues

* Blocks to #2768 
* Blocked by #2789
